src/tactic/core.lean

```diff
@@ -1144,8 +1144,17 @@ do e ← get_env,
   since it is expensive to execute `get_mathlib_dir` many times. -/
 meta def is_in_mathlib (n : name) : tactic bool :=
 do ml ← get_mathlib_dir, e ← get_env, return $ e.is_prefix_of_file ml n
+
+private meta def apply_under_pis_aux (p : pexpr) (cnst : pexpr) : ℕ → expr → pexpr
```

Comment:
Even though these definitions are private, a docstring still be useful so that people understand what is going on.

Also, I would personally do these kinds of things with a combination of `mk_local_pis`, `pis` and `expr.mk_app`/`tactic.mk_app`/`tactic.mk_mapp`, so that I don't have to deal with a combination of `expr` and `pexpr`s. I'm not saying you should change this, but I just thought I would mention.

Reply:
> Even though these definitions are private, a docstring still be useful so that people understand what is going on.

A taste of my own medicine ;) will do.

> I'm not saying you should change this, but I just thought I would mention.

It took a bit of trial and error to handle universes correctly, which is how I ended up with the current solution. (The `under` and `over` category instances were good tests for this.) Now that I see this solution maybe I can do it properly with those functions.